app/services/statements/statements_query.rb

```diff
@@ -0,0 +1,28 @@
+module Statements
+  class StatementsQuery
+    def initialize(lead_provider:, params: {})
+      @lead_provider = lead_provider
+      @cohorts = String(params[:cohort]).split(",")
+      @updated_since = params[:updated_since]
+    end
+
+    def statements
+      statements = Statement.includes(:cohort).where(lead_provider:)
+      statements = statements.where(cohorts: { start_year: cohorts }) if cohorts.any?
+      statements = statements.where("updated_at >= ?", @updated_since) if @updated_since.present?
+
+      statements
+    end
+
+    def statement(id)
+      Statement.find_by!(id:, lead_provider:)
+    end
```

Comment:
in ECF we have pattern of passing in the `id` with `params`:
```ruby
    def statement
      Statement.find_by!(id: params[:id], lead_provider:)
    end
```

Reply:
You wouldn't have access to `params` in a service class through? 🤔

Reply:
Maybe we can use explicit naming when initialising this service, rather than passing in params?